qiskit/circuit/library/standard_gates/x.py

```diff
@@ -569,6 +569,20 @@ def _define(self):
 
         self.definition = qc
 
+    def qasm(self):
+        # Gross hack to override the Qiskit name with the name this gate has in Terra's version of
+        # 'qelib1.inc'.  In general, the larger exporter mechanism should know about this to do the
+        # mapping itself, but right now that's not possible without a complete rewrite of the OQ2
+        # exporter code (low priority), or we would need to modify 'qelib1.inc' which would be
+        # needlessly disruptive this late in OQ2's lifecycle.  The current OQ2 exporter _always_
+        # outputs the `include 'qelib1.inc' line.  ---Jake, 2022-11-21.
+        try:
+            old_name = self.name
+            self.name = "c3sqrtx"
+            return super().qasm()
+        finally:
+            self.name = old_name
```

Comment:
The name is hardcoded for the class. What's the point of saving/restoring it? Does the user has any way to change it? If so, maybe a test should cover that case.

Reply:
The hardcoding is because it's hardcoded in `qelib1.inc`?  There's no valid reason a user should change that.  It needs to be saved/restored because the `qasm()` method requires it to be on the class like this, but Qiskit requires it to be `c3sx` because that's how we define the gate.  Since the OQ2 "exporter" isn't really one contained piece of logic, the instance itself has to mutate temporarily to ensure the correct thing gets output.

The alternative to this is completely rewriting the interface of the exporter, which while it should be done in the future, is quite beyond the scope of this bugfix.  This way keeps the bugfix much more local - it's explained in the comment above these lines.